dpnp/tests/test_product.py

```diff
@@ -1181,6 +1184,36 @@ def test_special_case(self, dt_out, shape1, shape2):
         result = dpnp.matmul(ia, ib, out=iout)
         assert_dtype_allclose(result, expected)
 
+    @pytest.mark.parametrize("dt", get_float_complex_dtypes())
+    def test_syrk(self, dt):
+        a = generate_random_numpy_array((6, 9), dtype=dt)
+        ia = dpnp.array(a)
+
+        result = dpnp.matmul(ia, ia.mT)
```

Comment:
We probably need to check also `dpnp.matmul(ia.mT, ia)` or shortcut `ia.mT @ ia`